docs/modules/ROOT/pages/how-to-guides/testing_applications/enable_snyk_check_for_a_product.adoc

```diff
@@ -0,0 +1,81 @@
+Specifically, the Snyk check scans an application's source code for potential security vulnerabilities, 
+including SQL injection, cross-site scripting (XSS), and code injection attack vulnerabilities.
+
+> NOTE: You can run a Snyk task only if you have a Snyk token stored in a namespace secret. 
+This must also include the name of your secret in the *snyk-secret* pipeline parameter.
+
+== Enabling Snyk task
+
+. Register for a Snyk account or log in at https://app.snyk.io/.
+. Get a Snyk token.
+.. In the lower left of the home page, click your name, then select *Account* settings.
+.. From the Account Settings page, select *General*, which is the default, then *Auth Token*.
+.. Under the *Auth Token* section, click on *Click to View* button to visualize the *KEY* value of the automatically generated token.
+. To enable Snyk code, go to *Settings* on the left panel > *Snyk Code* > *Enable Snyk Code* > click the *Disabled* bar.
+.. Save the changes when it is changed to *Enabled*.
+. Add the above generated secret *KEY* value to your {ProductName} workspace.
+    .. Login to the {ProductName} UI console https://console.redhat.com/preview/application-pipeline.
+    .. Click on the *Secrets* from the left panel.
+    .. Click on the *Add secret* button.
+    .. Fill out the details on the page appeared.
+    ... Select Secret for *Build*.
+    ... From the *Secret type* dropdown menu, choose *Key/value secret*.
+    ... From the *Secret name* dropdown menu, select *snyk-secret*.
+    ... Under *Key/value secret*, expand *Key/value 1* section the *Key* name is automatically populated as *snyk-token*.
+    ... Paste *snyk-token* value generated from the Snyk account in the above steps into the *Upload the file with value for your key or paste its contents* section.
+    ... Click on the *Add secret* button on the bottom of the page to save it.
+. Update .tekton yaml file(s) to specify the Snyk secret name.
+.. From your github repo, update the pull-request.yaml file(s) under *.tekton* directory, example: link:https://github.com/sonam1412/devfile-sample-python-basic/blob/main/.tekton/devfile-sample-python-basic-iea1-pull-request.yaml[devfile-sample-python-basic/.tekton/devfile-sample-python-basic-rhq8-pull-request.yaml]
+.. Add new lines to the code to include *snyk-secret* as suggested below:
++ 
+Example:
++
+[source]
+--
+    - default: "false"
+      description: Execute the build with network isolation
+      name: hermetic
+      type: string
+    - default: ""
+      description: Build dependencies to be prefetched by Cachi2
+      name: prefetch-input
+      type: string
+    - default: "false"
+      description: Java build
+      name: java
+      type: string
+    - default: "snyk-secret"
+      description: Snyk Token Secret Name
+      name: snyk-secret
+      type: string
+--
+
+You've enabled the Snyk task for your build pipeline.
+
+== Obtaining a Red Hat Snyk token using a service account
```

Comment:
@sonam1412 i'll have to get to this tomorrow if that's ok!

Reply:
Thank you so much for the reviews so far and helping me shape this document better. I have addressed all the review comments so far. 
Handing it over to you for the next set of reviews.